src/darwin/Framework/CHIPTests/TestHelpers/MTRTestKeys.m

```diff
@@ -31,6 +31,13 @@ - (NSData *)publicKeyData
     return (__bridge_transfer NSData *) SecKeyCopyExternalRepresentation([self publicKey], nil);
 }
 
+- (SecKeyRef)copyPublicKey {
+    // because this varies significantly from the actual implementations,
+    // it's probably not a great idea to rely on this for memory leak behavior information.
```

Comment:
I don't get this comment.  The implementation here ("I have a key, and I will CFRetain it and return it to you") is _exactly_ what a "normal" MTRKeyPair implementation will do.